### ISSUE TITLE
error on setup

### ISSUE THREAD

**Issue by fyee** [unindented/electron-installer-windows#24]
Creating package (this may take a while)
[Error: Error creating package: Error executing file (spawn mono ENOENT): 
mono /usr/local/lib/node_modules/electron-installer-windows/vendor/nuget/NuGet.exe pack /var/folders/z_/yxr814_533d817mfcglr1njr0000gp/T/electron-116323-1825-k26tmn/e3c_1.0.0/nuget/e3c.nuspec -BasePath /var/folders/z_/yxr814_533d817mfcglr1njr0000gp/T/electron-116323-1825-k26tmn/e3c_1.0.0/e3c -OutputDirectory /var/folders/z_/yxr814_533d817mfcglr1njr0000gp/T/electron-116323-1825-k26tmn/e3c_1.0.0/nuget -NoDefaultExcludes
] 'Error: Error creating package: Error executing file (spawn mono ENOENT): \nmono /usr/local/lib/node_modules/electron-installer-windows/vendor/nuget/NuGet.exe pack /var/folders/z_/yxr814_533d817mfcglr1njr0000gp/T/electron-116323-1825-k26tmn/e3c_1.0.0/nuget/e3c.nuspec -BasePath /var/folders/z_/yxr814_533d817mfcglr1njr0000gp/T/electron-116323-1825-k26tmn/e3c_1.0.0/e3c -OutputDirectory /var/folders/z_/yxr814_533d817mfcglr1njr0000gp/T/electron-116323-1825-k26tmn/e3c_1.0.0/nuget -NoDefaultExcludes\n\n    at /usr/local/lib/node_modules/electron-installer-windows/src/installer.js:273:21\n    at ChildProcess.<anonymous> (/usr/local/lib/node_modules/electron-installer-windows/src/installer.js:62:5)\n    at emitTwo (events.js:105:20)\n    at ChildProcess.emit (events.js:185:7)\n    at maybeClose (internal/child_process.js:850:16)\n    at Socket.<anonymous> (internal/child_process.js:323:11)\n    at emitOne (events.js:90:13)\n    at Socket.emit (events.js:182:7)\n    at Pipe._onclose (net.js:477:12)'

**Comment by unindented** [comment]
The error message implies that you don't have `mono` installed. Please read the `README.md`, it states the system requirements at the very beginning.

**Comment by austinmayer** [comment]
I have `mono` and `wine` installed via brew and got a similar error when building on OSX for windows.

**Comment by unindented** [comment]
@fyee @austinmayer, just tried building the example app on OSX with Node 4.4.3 and things work fine, and Travis CI builds are green, so this suggests an environment problem.

Can you clone this repository, `cd` into the `example` directory, and run the following?

```
$ DEBUG='electron-installer-windows' npm run build
```

@austinmayer what does `similar` mean? Is it the same error or not? Can you paste the output of the command above? Also, please specify which version of Node and npm you are using.

**Comment by unindented** [comment]
Closing for now. You can reopen if you provide repro steps.

**Comment by eddievagabond** [comment]
I had this same exact issue. For me, the issue was solved by insuring the name in my package.json file was the same as the exe file name to be package. For example: 

package.json
```
{
  "name": "someApp",
  ...
  "scripts": {
        "package": "electron-packager . someApp --overwrite --platform=win32 --arch=x64 --asar=true --out=./release",
  }
}
```

**Comment by phoniks** [comment]
I ran into this error as well.  Here's the output of the DEBUG that @unindented requested others:

````
Johns-MacBook-Pro-2:example johnhopkins-lg$ DEBUG='electron-installer-windows' npm run build

> poopie@0.0.1 build /Users/johnhopkins-lg/electron-installer-windows/example
> npm run clean && npm run exe32 && npm run set32 && npm run exe64 && npm run set64


> poopie@0.0.1 clean /Users/johnhopkins-lg/electron-installer-windows/example
> rimraf dist


> poopie@0.0.1 exe32 /Users/johnhopkins-lg/electron-installer-windows/example
> electron-packager . poopie --platform win32 --arch ia32 --out dist/ --icon resources/Icon.ico

Downloading tmp-14594-0-electron-v4.0.1-win32-ia32.zip
[============================================>] 100.0% of 58.1 MB (11.07 MB/s)
Packaging app for platform win32 ia32 using electron v4.0.1
Wrote new app to dist/poopie-win32-ia32

> poopie@0.0.1 set32 /Users/johnhopkins-lg/electron-installer-windows/example
> electron-installer-windows --src dist/poopie-win32-ia32/ --dest dist/installers/ia32/ --config config.json

Creating package (this may take a while)
  electron-installer-windows Reading package metadata from dist/poopie-win32-ia32/resources/app/package.json +0ms
  electron-installer-windows Creating package with options
  electron-installer-windows {
  electron-installer-windows   "help": false,
  electron-installer-windows   "src": "dist/poopie-win32-ia32/",
  electron-installer-windows   "dest": "dist/installers/ia32/",
  electron-installer-windows   "config": "config.json",
  electron-installer-windows   "icon": "resources/Icon.ico",
  electron-installer-windows   "tags": [
  electron-installer-windows     "Utility"
  electron-installer-windows   ],
  electron-installer-windows   "name": "poopie",
  electron-installer-windows   "productName": "poopie",
  electron-installer-windows   "description": "An example app, built with Electron.",
  electron-installer-windows   "productDescription": "An example app, built with Electron.",
  electron-installer-windows   "version": "0.0.1",
  electron-installer-windows   "copyright": "Copyright © 2019 Daniel Perez Alvarez",
  electron-installer-windows   "authors": [
  electron-installer-windows     "Daniel Perez Alvarez"
  electron-installer-windows   ],
  electron-installer-windows   "owners": [
  electron-installer-windows     "Daniel Perez Alvarez"
  electron-installer-windows   ],
  electron-installer-windows   "homepage": "http://unindented.org/",
  electron-installer-windows   "exe": "poopie.exe",
  electron-installer-windows   "animation": "/Users/johnhopkins-lg/electron-installer-windows/example/node_modules/electron-installer-windows/resources/animation.gif",
  electron-installer-windows   "requireLicenseAcceptance": false,
  electron-installer-windows   "noMsi": false
  electron-installer-windows } +4ms
  electron-installer-windows Creating temporary directory +1ms
  electron-installer-windows Creating subdirectories under /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1 +40ms
  electron-installer-windows Creating contents of package +1ms
  electron-installer-windows Creating spec file at /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1/nuget/poopie.nuspec +0ms
  electron-installer-windows Generating template from /Users/johnhopkins-lg/electron-installer-windows/example/node_modules/electron-installer-windows/resources/spec.ejs +1ms
  electron-installer-windows Copying application to /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1/poopie +0ms
  electron-installer-windows Generated template from /Users/johnhopkins-lg/electron-installer-windows/example/node_modules/electron-installer-windows/resources/spec.ejs
  electron-installer-windows <?xml version="1.0" encoding="utf-8"?>
  electron-installer-windows <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
  electron-installer-windows   <metadata>
  electron-installer-windows     <id>poopie</id>
  electron-installer-windows     <title>poopie</title>
  electron-installer-windows     <summary>An example app, built with Electron.</summary>
  electron-installer-windows     <description>An example app, built with Electron.</description>
  electron-installer-windows     <version>0.0.1</version>
  electron-installer-windows     <copyright>Copyright © 2019 Daniel Perez Alvarez</copyright>
  electron-installer-windows     <authors>Daniel Perez Alvarez</authors>
  electron-installer-windows     <owners>Daniel Perez Alvarez</owners>
  electron-installer-windows     <projectUrl>http://unindented.org/</projectUrl>
  electron-installer-windows
  electron-installer-windows
  electron-installer-windows
  electron-installer-windows     <tags>Utility</tags>
  electron-installer-windows   </metadata>
  electron-installer-windows   <files>
  electron-installer-windows     <file src="locales\**" target="lib\net45\locales" />
  electron-installer-windows     <file src="resources\**" target="lib\net45\resources" />
  electron-installer-windows     <file src="*.bin" target="lib\net45" />
  electron-installer-windows     <file src="*.dll" target="lib\net45" />
  electron-installer-windows     <file src="*.pak" target="lib\net45" />
  electron-installer-windows     <file src="Update.exe" target="lib\net45\squirrel.exe" />
  electron-installer-windows     <file src="icudtl.dat" target="lib\net45\icudtl.dat" />
  electron-installer-windows     <file src="LICENSE" target="lib\net45\LICENSE" />
  electron-installer-windows     <file src="poopie.exe" target="lib\net45\poopie.exe" />
  electron-installer-windows   </files>
  electron-installer-windows </package>
  electron-installer-windows  +7ms
  electron-installer-windows Creating package at /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1 +893ms
  electron-installer-windows Executing command: mono /Users/johnhopkins-lg/electron-installer-windows/example/node_modules/electron-installer-windows/vendor/nuget/NuGet.exe pack /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1/nuget/poopie.nuspec -BasePath /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1/poopie -OutputDirectory /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1/nuget -NoDefaultExcludes +1ms
  electron-installer-windows Error creating package: Error creating package with NuGet: Error executing command (spawn mono ENOENT):
  electron-installer-windows mono /Users/johnhopkins-lg/electron-installer-windows/example/node_modules/electron-installer-windows/vendor/nuget/NuGet.exe pack /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1/nuget/poopie.nuspec -BasePath /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1/poopie -OutputDirectory /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1/nuget -NoDefaultExcludes
  electron-installer-windows  +14ms
Error: Error creating package with NuGet: Error executing command (spawn mono ENOENT):
mono /Users/johnhopkins-lg/electron-installer-windows/example/node_modules/electron-installer-windows/vendor/nuget/NuGet.exe pack /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1/nuget/poopie.nuspec -BasePath /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1/poopie -OutputDirectory /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1/nuget -NoDefaultExcludes

    at err (/Users/johnhopkins-lg/electron-installer-windows/example/node_modules/electron-installer-windows/src/installer.js:31:11)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7) 'Error: Error creating package with NuGet: Error executing command (spawn mono ENOENT):\nmono /Users/johnhopkins-lg/electron-installer-windows/example/node_modules/electron-installer-windows/vendor/nuget/NuGet.exe pack /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1/nuget/poopie.nuspec -BasePath /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1/poopie -OutputDirectory /var/folders/ky/8mxbkmwd5q32k4l4fd1b23nm0000gn/T/electron-14649Axwr5fOCbdCS/poopie_0.0.1/nuget -NoDefaultExcludes\n\n    at err (/Users/johnhopkins-lg/electron-installer-windows/example/node_modules/electron-installer-windows/src/installer.js:31:11)\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:188:7)'
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! poopie@0.0.1 set32: `electron-installer-windows --src dist/poopie-win32-ia32/ --dest dist/installers/ia32/ --config config.json`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the poopie@0.0.1 set32 script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/johnhopkins-lg/.npm/_logs/2019-01-13T05_33_52_709Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! poopie@0.0.1 build: `npm run clean && npm run exe32 && npm run set32 && npm run exe64 && npm run set64`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the poopie@0.0.1 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/johnhopkins-lg/.npm/_logs/2019-01-13T05_33_52_758Z-debug.log
````

**Comment by fcastilloec** [comment]
@phoniks Please don't comment in a closed issue when it has been fixed. As clearly stated by the first response, your problem is that you don't have `mono` installed. Just run `brew install mono`. All this info was clearly stated in the README

**Comment by phoniks** [comment]
@fcastilloec - I'm sorry for any inconvenience my comment may have caused, but I think it has resulted in a productive discussion that may be of value to others who run into this issue in the future.  To be clear, none of the previous answers mentioned that installing mono through homebrew was an option.  I tried to install from the .pkg on their website, but the command still failed. So, if you've found your way here because you've got this error, and installing mono via the methods listed at mono-project.com doesn't work try installing via homebrew and it should work.  

bottom line:
**INSTALL mono via homebrew**
